src/runtime_src/core/edge/drm/zocl/zocl_cu_xgq.c

```diff
@@ -404,37 +404,22 @@ int zcu_xgq_unassign_cu(struct platform_device *pdev, u32 cu_idx, u32 cu_domain)
 	return rc;
 }
 
-static void init_resp(struct xgq_com_queue_entry *resp, u16 cid, u32 rcode)
-{
-	memset(resp, 0, sizeof(*resp));
-	resp->hdr.cid = cid;
-	resp->hdr.cstate = XGQ_CMD_STATE_COMPLETED;
-	resp->result = KDS_COMPLETED;
-	resp->rcode = rcode;
-}
-
-static void init_resp_abort(struct xgq_com_queue_entry *resp, u16 cid, u32 rcode, int status)
-{
-	memset(resp, 0, sizeof(*resp));
-	resp->hdr.cid = cid;
-	resp->hdr.cstate = XGQ_CMD_STATE_ABORTED;
-	resp->result = status;
-	resp->rcode = rcode;
-}
-
 static inline void
 zcu_xgq_cmd_complete(struct platform_device *pdev, struct xgq_cmd_sq_hdr *cmd, int ret, int status)
 {
-	struct xgq_com_queue_entry r;
-	struct xgq_com_queue_entry *rptr = &r;
+	struct xgq_com_queue_entry r = {0};
+	struct xgq_com_queue_entry *rptr = NULL;
 	struct zocl_cu_xgq *zcu_xgq = platform_get_drvdata(pdev);
 
-	if (likely(ret == 0 && ZCU_XGQ_FAST_PATH(zcu_xgq)))
-		rptr = NULL;
-	else if (unlikely(status == KDS_SKCRASHED))
-		init_resp_abort(&r, cmd->cid, ret, status);
-	else
-		init_resp(&r, cmd->cid, ret);
+	if (unlikely(!(ret == 0 && ZCU_XGQ_FAST_PATH(zcu_xgq)))) {
+		rptr = &r;
+		r.hdr.cid = cmd->cid;
+		r.hdr.cstate = XGQ_CMD_STATE_COMPLETED;
+		r.result = status;
+		r.rcode = ret;
+		if (unlikely(status == KDS_SKCRASHED))
```

Comment:
`r.hdr.cstate = (status == KDS_SKCRASHED) ? XGQ_CMD_STATE_ABORTED : XGQ_CMD_STATE_COMPLETED`

and remove the previous r.hdr.cstate assignment.